lib/state.js

```diff
@@ -872,32 +872,46 @@ class State {
     assert(result.rowCount === 1, 'inserting spot request had incorrect rowCount');
   }
 
-  async reportEbsUsage({region, volumetype, state, totalcount, totalgb}, client) {
-    assert(typeof region === 'string');
-    assert(typeof volumetype === 'string');
-    assert(typeof state === 'string');
-    assert(typeof totalcount === 'number');
-    assert(typeof totalgb === 'number');
+  async reportEbsUsage(rowData) { //, client) {
+    assert(typeof rowData === 'object');
+    assert(Array.isArray(rowData));
+
+    let transaction = await this.beginTransaction();
```

Comment:
This is good, but needs a try/catch to roll the transaction back if something goes wrong.